### PR TITLE
Fixed overlays closing while still dragging

### PR DESCRIPTION
**Pull request by bramvanderholst**
Overlays should only close after releasing pointer

### CODE REVIEW

File: packages/next-ui/Overlay/components/OverlayBase.tsx [hunk 1]
```diff
@@ -311,10 +312,34 @@ export function OverlayBase(incomingProps: LayoutOverlayBaseProps) {
   }
   useDomEvent(windowRef, 'keyup', handleEscape, { passive: true })
 
+  let dragging = false
+  const backdropRef = useRef<HTMLDivElement>(null)
+  const scrollerElementRef = scrollerRef as React.RefObject<HTMLElement>
+
+  const handleDragStart = () => {
+    dragging = true
```
Comment (paales) [inline packages/next-ui/Overlay/components/OverlayBase.tsx:320]:
```suggestion
    dragging.set(true)
```

File: packages/next-ui/Overlay/components/OverlayBase.tsx [hunk 2]
```diff
@@ -311,10 +312,34 @@ export function OverlayBase(incomingProps: LayoutOverlayBaseProps) {
...
+  }
+  const handleDragEnd = () => {
+    dragging = false
```
Comment (paales) [inline packages/next-ui/Overlay/components/OverlayBase.tsx:323]:
```suggestion
    dragging.set(false)
```

File: packages/next-ui/Overlay/components/OverlayBase.tsx [hunk 3]
```diff
@@ -311,10 +312,34 @@ export function OverlayBase(incomingProps: LayoutOverlayBaseProps) {
...
+  useDomEvent(scrollerElementRef, 'mouseup', handleDragEnd, { passive: true })
+
   // When the overlay isn't visible anymore, we navigate back.
-  useEffect(
-    () => positions.open.visible.on('change', (o) => o === 0 && closeOverlay()),
-    [closeOverlay, position, positions.open.visible],
+  useMotionValueEvent(
+    positions.open.visible,
+    'change',
+    (o) => o === 0 && !dragging && closeOverlay(),
```
Comment (paales) [inline packages/next-ui/Overlay/components/OverlayBase.tsx:342]:
```suggestion
    (o) => o === 0 && !dragging.get() && closeOverlay(),
```

**Review by paales — CHANGES_REQUESTED** [review]

**Comment by bramvanderholst** [comment]
@paales The requested changes have been implemented.

**Pull request merged**